### ISSUE TITLE
Why did you comment out the softmax in `2. BiDAF.ipynb`?

### ISSUE THREAD

**Issue by shahhaard47** [kushalj001/pytorch-question-answering#4]
This is at the end of the **BiDAF.forward**

```Python
        p1 = p1.squeeze()
        # [bs, ctx_len]
        
        #p1 = F.softmax(p1, dim=-1)
        
        # END PREDICTION
        
        p2 = self.output_end(torch.cat([G, M2], dim=2)).squeeze()
        # [bs, ctx_len, 1] => [bs, ctx_len]
        
        #p2 = F.softmax(p2, dim=-1)
```

**Comment by shahhaard47** [comment]
Also, I probably missed something in the paper, but why don't you include bias in any Linear layers? You do:
```Python
nn.Linear(..., bias=False)
```

**Comment by kushalj001** [comment]
> Also, I probably missed something in the paper, but why don't you include bias in any Linear layers? You do:
![image](https://user-images.githubusercontent.com/32245327/96618097-1f4eb600-1322-11eb-8b7a-5f720feb3888.png)

If you read the paper closely, you'll see that we usually only need a weight matrix and not a linear layer per se.  A linear layer acts as a weight matrix if you make `bias=False`. Alternatively, you could have also used the `nn.Parameter()` to initialize the weight matrix, so that it gets added to the list of model parameters. I think using `nn.Linear` reduces some boilerplate code.

**Comment by kushalj001** [comment]
> This is at the end of the **BiDAF.forward**

I don't use softmax in the model because I use `nn.CrossEntropy`  to calculate the loss which takes care of all that internally.

**Comment by shahhaard47** [comment]
Thank you so much for the response!
